go-controller/pkg/ovn/ovn.go

```diff
@@ -524,12 +529,22 @@ func (oc *Controller) ensurePod(oldPod, pod *kapi.Pod, addPort bool) bool {
 }
 
 // WatchPods starts the watching of Pod resource and calls back the appropriate handler logic
-func (oc *Controller) WatchPods() {
+func (oc *Controller) WatchPods(wg *sync.WaitGroup) {
 	go func() {
 		// track the retryPods map and every 30 seconds check if any pods need to be retried
 		utilwait.Until(oc.iterateRetryPods, 30*time.Second, oc.stopChan)
 	}()
 
+	// start podSetController
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		err := oc.podSetController.Run(5, oc.stopChan)
+		if err != nil {
+			klog.Errorf("Error running PodSet controller: %v")
+		}
+	}()
```

Comment:
All this logic should be added to `podSetController.Run()`, the ovn controller should only call `err:=podSetController.Run(5,oc.stopChan)`